src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -153,25 +153,30 @@ internal void Abort()
         {
             if (!_isClosed)
             {
-                lock (this)
+                try
                 {
-                    if (!_isClosed && _tcpClient != null)
+                    lock (this)
                     {
-                        //free CBT buffer
-                        if (_channelBindingToken != null)
+                        if (!_isClosed && _tcpClient != null)
                         {
-                            _channelBindingToken.Close();
+                            // free CBT buffer
+                            _channelBindingToken?.Close();
+
+                            // must destroy manually since sending a QUIT here might not be
```

Comment:
I know you didn't write this comment, but since you're moving it anyway...
```suggestion
                            // Must destroy manually since sending a QUIT here might not be
```